chia/consensus/blockchain.py

```diff
@@ -689,10 +689,20 @@ async def get_block_records_at(self, heights: List[uint32]) -> List[BlockRecord]
         """
         gets block records by height (only blocks that are part of the chain)
         """
+        records: List[BlockRecord] = []
         hashes = []
+
         for height in heights:
             hashes.append(self.height_to_hash(height))
-        return await self.block_store.get_block_records_by_hash(hashes)
+            if len(hashes) > 900:
```

Comment:
I think there should be a comment here, saying sqlite in python 3.7 has a limit on 999 variables in queries, and that's why we're batching